renderer/src/features/mcp-servers/components/customize-tools-table.tsx

```diff
@@ -8,26 +8,94 @@ import {
   TableHeader,
   TableRow,
 } from '@/common/components/ui/table'
-import { useState, useEffect } from 'react'
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from '@/common/components/ui/tooltip'
+import { useState, useEffect, useCallback } from 'react'
 import { Skeleton } from '@/common/components/ui/skeleton'
+import {
+  Alert,
+  AlertDescription,
+  AlertTitle,
+} from '@/common/components/ui/alert'
+import { Badge } from '@/common/components/ui/badge'
+import { TriangleAlert } from 'lucide-react'
 
 interface Tool {
   name: string
   description?: string
+  isInitialEnabled?: boolean
 }
 
 interface CustomizeToolsTableProps {
   tools: Tool[]
   isLoading?: boolean
-  onApply?: (enabledTools: Record<string, boolean>) => void
-  onCancel?: () => void
+  toolsDiff?: {
+    hasExactMatch: boolean
+    addedTools: string[]
+    missingTools: string[]
+  } | null
```

Comment:
it seems repeatable in multiple files, maybe it could be a shared type

Reply:
I added a interface for it